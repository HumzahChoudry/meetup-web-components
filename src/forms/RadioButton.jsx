import React from 'react';
import PropTypes from 'prop-types';

import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';

const RadioButton = ({ label, className, ...inputProps }) => {
	const id = `RadioButton-${inputProps.name}-${inputProps.value}`;

	return (
		<Flex align="center" className={className} noGutters>
			<FlexItem shrink>
				<input readOnly type="radio" id={id} {...inputProps} />
			</FlexItem>
			<FlexItem shrink>
				<label htmlFor={id}>
					{label}
				</label>
			</FlexItem>
		</Flex>
	);
};

RadioButton.displayName = 'RadioButton';

RadioButton.propTypes = {
	label: PropTypes.string.isRequired,
	className: PropTypes.string,
};

export default RadioButton;
