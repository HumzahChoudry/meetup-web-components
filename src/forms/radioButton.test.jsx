import React from 'react';
import { shallow } from 'enzyme';

import RadioButton from './RadioButton';

describe('RadioButton', () => {
	it('should match the snapshot', () => {
		const props = {
			label: 'This is a label',
			className: 'some-class',
			checked: true,
			otherProp: 'other',
			value: 'value',
			name: 'name',
		};
		expect(shallow(<RadioButton {...props} />)).toMatchSnapshot();
	});
});
