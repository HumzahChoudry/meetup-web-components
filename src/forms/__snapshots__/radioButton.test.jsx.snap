// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton checked should render an disabled radio button 1`] = `
<label
  className="toggleLabel label--minor display--block"
  htmlFor="RadioButton-name-value"
>
  <WithLoading(Flex)
    align="center"
    className="some-class"
    noGutters={true}
  >
    <WithLoading(FlexItem)
      shrink={true}
    >
      <input
        checked={true}
        className="radio visibility--a11yHide"
        disabled={false}
        id="RadioButton-name-value"
        name="name"
        otherProp="other"
        readOnly={true}
        type="radio"
        value="value"
      />
      <span
        className="display--flex flex--alignCenter flex--center align--center fauxToggle fauxToggle--radio checked"
      >
        <span
          className="radio-indicator"
        />
      </span>
    </WithLoading(FlexItem)>
    <WithLoading(FlexItem)
      className="toggleLabel-container"
      shrink={true}
    >
      <span
        className="text--bold"
      >
        This is a label
      </span>
    </WithLoading(FlexItem)>
  </WithLoading(Flex)>
</label>
`;

exports[`RadioButton checked should render an enabled radio button 1`] = `
<label
  className="toggleLabel label--minor display--block"
  htmlFor="RadioButton-name-value"
>
  <WithLoading(Flex)
    align="center"
    className="some-class"
    noGutters={true}
  >
    <WithLoading(FlexItem)
      shrink={true}
    >
      <input
        checked={true}
        className="radio visibility--a11yHide"
        disabled={true}
        id="RadioButton-name-value"
        name="name"
        otherProp="other"
        readOnly={true}
        type="radio"
        value="value"
      />
      <span
        className="display--flex flex--alignCenter flex--center align--center fauxToggle fauxToggle--radio checked disabled"
      >
        <span
          className="radio-indicator"
        />
      </span>
    </WithLoading(FlexItem)>
    <WithLoading(FlexItem)
      className="toggleLabel-container"
      shrink={true}
    >
      <span
        className="text--hint text--bold"
      >
        This is a label
      </span>
    </WithLoading(FlexItem)>
  </WithLoading(Flex)>
</label>
`;

exports[`RadioButton unchecked should render an disabled radio button 1`] = `
<label
  className="toggleLabel label--minor display--block"
  htmlFor="RadioButton-name-value"
>
  <WithLoading(Flex)
    align="center"
    className="some-class"
    noGutters={true}
  >
    <WithLoading(FlexItem)
      shrink={true}
    >
      <input
        checked={false}
        className="radio visibility--a11yHide"
        disabled={false}
        id="RadioButton-name-value"
        name="name"
        otherProp="other"
        readOnly={true}
        type="radio"
        value="value"
      />
      <span
        className="display--flex flex--alignCenter flex--center align--center fauxToggle fauxToggle--radio"
      />
    </WithLoading(FlexItem)>
    <WithLoading(FlexItem)
      className="toggleLabel-container"
      shrink={true}
    >
      <span
        className=""
      >
        This is a label
      </span>
    </WithLoading(FlexItem)>
  </WithLoading(Flex)>
</label>
`;

exports[`RadioButton unchecked should render an enabled radio button 1`] = `
<label
  className="toggleLabel label--minor display--block"
  htmlFor="RadioButton-name-value"
>
  <WithLoading(Flex)
    align="center"
    className="some-class"
    noGutters={true}
  >
    <WithLoading(FlexItem)
      shrink={true}
    >
      <input
        checked={false}
        className="radio visibility--a11yHide"
        disabled={true}
        id="RadioButton-name-value"
        name="name"
        otherProp="other"
        readOnly={true}
        type="radio"
        value="value"
      />
      <span
        className="display--flex flex--alignCenter flex--center align--center fauxToggle fauxToggle--radio disabled"
      />
    </WithLoading(FlexItem)>
    <WithLoading(FlexItem)
      className="toggleLabel-container"
      shrink={true}
    >
      <span
        className="text--hint"
      >
        This is a label
      </span>
    </WithLoading(FlexItem)>
  </WithLoading(Flex)>
</label>
`;
